govcd/catalog_test.go

```diff
@@ -1490,3 +1490,40 @@ func (vcd *TestVCD) Test_CatalogCreateCompleteness(check *C) {
 	err = catalog.Delete(true, true)
 	check.Assert(err, IsNil)
 }
+
+func (vcd *TestVCD) Test_CaptureVapp(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	vapp, vm := createNsxtVAppAndVm(vcd, check)
+	check.Assert(vapp, NotNil)
+	check.Assert(vm, NotNil)
+
+	// retrieve NSX-T Catalog
+	cat, err := vcd.org.GetCatalogByName(vcd.config.VCD.Catalog.NsxtBackedCatalogName, false)
+	check.Assert(err, IsNil)
+	check.Assert(cat, NotNil)
+
+	vAppCaptureParams := &types.CaptureVAppParams{
+		Name: check.TestName() + "vm-template",
+		Source: &types.Reference{
+			HREF: vapp.VApp.HREF,
+		},
+		CustomizationSection: types.CaptureVAppParamsCustomizationSection{
+			Info:                   "CustomizeOnInstantiate Settings",
+			CustomizeOnInstantiate: true,
+		},
+	}
+
+	if vcd.client.Client.APIVCDMaxVersionIs(">= 37.2") {
```

Comment:
I was having the idea of "hiding" this logic inside `CaptureVappTemplate` so it is done "automatically", but not sure if that would be "too much" (I mean, if <37.2 returns a meaningful error maybe it's not so convenient to hide it)

Reply:
I am up for both options. The error is not very beautiful, but it tells that it is an unrecognised field. I could migrate it.

Reply:
Changed.